src/repositories/AuthorRepository.ts

```diff
@@ -30,4 +30,6 @@ export default interface AuthorRepository {
   find(params: FindParameters): Promise<Author[]>;
 
   count(params: CountParameters): Promise<number>;
+
+  delete(id: number): Promise<number>;
```

Comment:
Here is, that the delete function is returning number. But in mutation, the return type is `AuthorType`.

Reply:
Maybe you can do:

```ts
delete(id: number): Pormise<Author> {
  const author = await this.get(id);

  await database.table('author').where('id', id).del();

  return author;
}
```